### PR TITLE
Minor UI/UX tweaks

### PR DESCRIPTION
**Pull request by bnjmnt4n**
This PR adds a couple of minor UI/UX tweaks:
- Code editor: add styling for the remote cursor
  - This adds an orange cursor to make it clearer where the other user is.
- Registering as a new user now logs you in as well. I think this behavior is a lot more intuitive.
- Login/register forms: use `<form>` elements to enable "Enter" for submission
- Header/account menu: tweak placement of the menu to be align its right edge to the right edge of the button

This is stacked on top of #13.

### CODE REVIEW

File: frontend/src/pages/_app.tsx [hunk 1]
```diff
@@ -2,6 +2,8 @@ import type { AppProps } from "next/app";
 import { ChakraProvider } from "@chakra-ui/react";
 import { AuthProvider } from "@/utils/auth";
 
+import "@/components/CodeRoom/CodeEditor/CodeEditor.css";
```
Comment (sebbycake) [inline frontend/src/pages/_app.tsx:5]: I haven't had the chance to pull the PR and run the code to test. It's possible that I've missed something, but I'm wondering if the CSS classes in `CodeEditor.css` are being used?

Comment (bnjmnt4n) [reply frontend/src/pages/_app.tsx:5]: Here's what it looks like following this addition:

Screenshot 2023-11-09 at 10 15 19 PM

I got the CSS from the yjs-monaco demo: https://github.com/yjs/y-monaco/blob/81e6d81c4ac2ba0e80ce3d176cc5bf0c76cec4d9/demo/index.html#L12-L30. I'm guessing it's a global classname applied by the library.

Comment (sebbycake) [reply frontend/src/pages/_app.tsx:5]: I see... thanks for explaining!

**Review by sebbycake — APPROVED** [review]
Thanks for looking into the remote cursor. I actually stated this as one of the suggestions for improvement in the report, but now i can remove it 😀

**Pull request merged**